code/modules/mining/machine_redemption.dm

```diff
@@ -500,6 +506,36 @@
 							"<span class='notice'>You insert [I] into [src].</span>")
 	return TRUE
 
+/obj/machinery/mineral/ore_redemption/proc/try_refill_storage(mob/living/silicon/robot/robot)
+	. = FALSE
+	if(!istype(robot))
+		return
+	if(!istype(robot.module, /obj/item/robot_module/engineering)) // Should only happen for drones
+		return
+
+	for(var/datum/robot_storage/material/mat_store in robot.module.material_storages)
+		if(mat_store.amount == mat_store.max_amount) // Already full, no need to run a check
+			to_chat(robot, "[mat_store] could not be filled due to it already being full.")
+			continue
+		var/datum/component/material_container/container_component = GetComponent(/datum/component/material_container)
+		for(var/mat_id as anything in container_component.materials)
+			var/datum/material/stack = container_component.materials[mat_id] // Should have only `/datum/material` in the list
+			var/obj/item/stack/sheet/sheet = stack.sheet_type
+			if(ispath(mat_store.stack, sheet))
+				var/amount_to_add
+				if(stack.amount >= (mat_store.max_amount - mat_store.amount))
+					amount_to_add = mat_store.max_amount - mat_store.amount
+					mat_store.amount = mat_store.max_amount
+					stack.amount -= amount_to_add * 2000 // The 2000 is to adjust for the sheets
```

Comment:
Can you make the `2000` magic number into a define?

Reply:
Yeah sounds good